integration/Substra/flamby-integration-into-substra.md

```diff
@@ -220,90 +160,118 @@ for ind, org_id in enumerate(ORGS_ID):
-    FedTcgaBrca,
-    Baseline,
-    BaselineLoss,
-    Optimizer,
-    BATCH_SIZE,
-    LR,
-)
+A metric is a function used to compute the score of predictions on one or several **datasamples**.
 
+To add a metric, you need to define a function that computes and return a performance from the datasamples (as returned by the opener) and the predictions_path (to be loaded within the function).
 
-model = Baseline()
-optimizer = Optimizer(model.parameters(), lr=LR)
-criterion = BaselineLoss()
-```
+When using a Torch SubstraFL algorithm, the predictions are saved in the predict function under the numpy format so that you can simply load them using `numpy.load`.
 
-## Substrafl imports
+After defining the metrics dependencies and permissions, we use the add_metric function to register the metric. This metric will be used on the test datasamples to evaluate the model performances.
 
 ```python
+import numpy as np
 
-from typing import Any
+from torch.utils import data
 
-from substrafl.algorithms.pytorch import TorchFedAvgAlgo
 from substrafl.dependency import Dependency
-from substrafl.strategies import FedAvg
-from substrafl.nodes import AggregationNode
-from substrafl.evaluation_strategy import EvaluationStrategy
-from substrafl.index_generator import NpIndexGenerator
-from substrafl.experiment import execute_experiment
+from substrafl.remote.register import add_metric
 
-import torch
 
-```
+def tgca_brca_metric(datasamples, predictions_path):
 
-## Substrafl algo definition
+    config = datasamples
 
-To instantiate a Substrafl [**Torch Algorithm**](https://docs.substra.org/en/0.21.0/substrafl_doc/api/algorithms.html#torch-algorithms), you need to define a torch Dataset with a specific `__init__` signature, that must contain (self, x, y, is_inference). This torch Dataset is useful to preprocess your data on the `__getitem__` function.
+    dataset = fed_tcga_brca.FedTcgaBrca(**config)
+    dataloader = data.DataLoader(dataset, batch_size=len(dataset))
 
-This dataset is passed **as a class** to the [**Torch Algorithms**](https://docs.substra.org/en/0.21.0/substrafl_doc/api/algorithms.html#torch-algorithms).
-Indeed, this torch Dataset will be instantiated within the algorithm, using the opener functions as x and y parameters.
+    y_true =  next(iter(dataloader))[1]
+    y_pred = np.load(predictions_path)
 
-The index generator will be used a the batch sampler of the dataset, in order to save the state of the seen samples during the training, as Federated Algorithms have a notion of `num_updates`, which forced the batch sampler of the dataset to be stateful.
+    return float(fed_tcga_brca.metric(y_true, y_pred))
 
-`NUM_UPDATES` is the number of model update between each FL strategy aggregation.
+# The Dependency object is instantiated in order to install the right libraries in
+# the Python environment of each organization.
+metric_deps = Dependency(pypi_dependencies=["torch==1.11.0","numpy==1.23.1"],
```

Comment:
explain local_dependencies in the comment, it's not clear to me what it does.

Reply:
Done :)